samples/extensions/open_cl_interop/open_cl_utils.cpp

```diff
@@ -0,0 +1,68 @@
+/* Copyright (c) 2021, Arm Limited and Contributors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "open_cl_utils.h"
+
+#include <dlfcn.h>
+#include <string>
+#include <vector>
+
+#define OPENCL_EXPORTED_FUNCTION(func_name) std::function<decltype(func_name)> func_name##_ptr = nullptr
+#define OPENCL_EXPORTED_EXTENSION_FUNCTION(func_name) std::function<decltype(func_name)> func_name##_ptr = nullptr
+#include "open_cl_functions.inl"
+
+static void *handle = nullptr;
+
+cl_platform_id load_opencl()
+{
+	static const std::vector<std::string> possible_libraries{"libOpenCL.so", "libGLES_mali.so", "libmali.so"};
+	for (const auto &possible_library : possible_libraries)
+	{
+		handle = dlopen("libOpenCL.so", RTLD_LAZY | RTLD_LOCAL);

```

Comment:
While this change is likely not needed, possible_library isn't used anywhere in the for loop; so dlopen libOpenCL.so is called 3 times.
I'd recommend reversing this change to just the single libOpenCL.so and allow ld to determine what dependencies it needs to load from OpenCL.

Reply:
Agree. I have reverted this change, thanks for your attentiveness!